### ISSUE TITLE
azurerm_data_factory_linked_custom_service using SnowflakeV2 does not support "role" property in type_properties_json

### ISSUE THREAD

**Issue by amcsi** [hashicorp/terraform-provider-azurerm#25317]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.5

### AzureRM Provider Version

3.94.0

### Affected Resource(s)/Data Source(s)

azurerm_data_factory_linked_custom_service

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  location = "East US"
  name     = "example"
}

resource "random_id" "example" {
  byte_length = 6
}

resource "azurerm_data_factory" "example" {
  location            = azurerm_resource_group.example.location
  name                = "adf-example-${random_id.example.id}"
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_data_factory_linked_custom_service" "example" {
  data_factory_id = azurerm_data_factory.example.id
  name            = "snowflake-example"

  type = "SnowflakeV2"
  type_properties_json = jsonencode({
    authenticationType = "Basic",
    accountIdentifier  = "something.east-us-2.azure",
    user               = "some_user",
    database           = "some_database",
    warehouse          = "some_warehouse",
    role               = "some_role",
  })
}
```

### Debug Output/Panic Output

```shell
(Success)

https://imgur.com/a/bkXARAp
```

### Expected Behaviour

The "Role" field should have the value "some_role" in it as per the `type_properties_json`.

Note: if you enter the "Role" field manually in the Portal and save it, then go look at the JSON of the linked service in the Portal, you'll see the `"role"` property appear among the "typeProperties"; this is evidence that the "role" property is supported by Azure (but apparently not by TF).

### Actual Behaviour

The "Role" field is empty.

### Steps to Reproduce

1. terraform apply

**Comment by npassmore** [comment]
Would be great if this could be fixed

**Comment by Dan-idbi** [comment]
Agreed, whenever you run a Terraform Apply against an environment with the SnowflakeV2 custom linked service you have to re-add the Role which isn't ideal

**Comment by ericsalesdeandrade** [comment]
++ 1 for this. 

Have to re-add the role and password for each snowflake DB linked service which is quite tedious if you have many.

**Comment by rcskosir** [comment]
:wave: Thanks for reaching out, unfortunately I do not have an ETA on this enhancement. Any future work via the team or the community should end up linked here via a PR.

**Comment by a-schot** [comment]
Waiting on this as well
